packages/identifier-resolution/src/functions/externalIdentifierFunctions.ts

```diff
@@ -214,6 +224,54 @@ export async function resolveExternalCoseKeyIdentifier(
   } satisfies ExternalIdentifierCoseKeyResult
 }
 
+export async function resolveExternalEntityIdIdentifier(
+  opts: ExternalIdentifierEntityIdOpts,
+  context: IAgentContext<IOIDFClient>
+): Promise<ExternalIdentifierEntityIdResult> {
+  let { trustAnchors, identifier } = opts
+
+  if (!trustAnchors || trustAnchors.length === 0) {
+    return Promise.reject(Error('ExternalIdentifierEntityIdOpts is missing the trustAnchors'))
+  }
+
+  if (!contextHasPlugin(context, 'jwtVerifyJwsSignature')) {
+    return Promise.reject(Error('For EntityId resolving the agent needs to have the JwtService plugin enabled'))
+  }
+
+  const trustedAnchors: Record<TrustedAnchor, PublicKeyHex> = {}
+  const errorList: Record<TrustedAnchor, ErrorMessage> = {}
+  const jwks: Array<ExternalJwkInfo> = []
+  
+  for (const trustAnchor of trustAnchors) {
+    const resolveResult = await context.agent.resolveTrustChain({
+      entityIdentifier: identifier,
+      trustAnchors: [trustAnchor]
+    })
+
+    if (resolveResult.error || !resolveResult.trustChain) {
+      errorList[trustAnchor] = resolveResult.errorMessage ?? 'unspecified'
+    } else {
+      for(const jwt of resolveResult.trustChain.asJsReadonlyArrayView()) {
+        const jwtVerifyResult = await context.agent.jwtVerifyJwsSignature({jwt})
+        if(jwtVerifyResult.error || jwtVerifyResult.critical) {
+          errorList[trustAnchor] = jwtVerifyResult.message
+          break
+        }
+        const esSignature = jwtVerifyResult.jws.signatures[0] // FIXME?
+        trustedAnchors[trustAnchor] = esSignature.publicKeyHex // When we have multiple hits from different trust anchors the caller can infer which signature came from which trust anchor  
+        jwks.concat(esSignature.identifier.jwks)
```

Comment:
I also expect a single Jwk that is the Jwk that the entity signed with. Also I would only expect the JWKS from the entity that we are looking up the trust for. The fact it has superiors which in turn have JWK(s) is a given. We obviously check their JWTs against their signatures. But from an application perspective I am only interested in the keys of the entity I am interacting with.

Reply:
fixed

Reply:
Please explain, as the authority is not what we are interested in in the end. We are interested in the keys of the entity

Reply:
fixed
const jwt = trustChain[trustChain.length - 1] => const jwt = trustChain[0]